### PR TITLE
Refactor code and update specs

### PR DESCRIPTION
**Pull request by johnmccrae**
…moved CRLF in favor of LF

Signed-off-by: John McCrae <john.mccrae@progress.com>

### Description

[Please describe what this change achieves]

### Issues Resolved

[List any existing issues this PR resolves, or any Discourse or
StackOverflow discussions that are relevant]

### Check List

- [ ] New functionality includes tests
- [ ] All tests pass
- [ ] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/main/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/win32/certstore/store_base.rb [hunk 1]
```diff
@@ -17,20 +17,28 @@
...
 require "json" unless defined?(JSON)
+require "chef-powershell"
+
+# 10-6-2021 - temporarily commenting out powershell_exec until it is moved to its own gem
+
+# begin
+#   require "chef/mixin/powershell_exec"
+# rescue LoadError
+#   puts "Not loading powershell_exec during testing"
+# end
```
Comment (lamont-granquist) [inline lib/win32/certstore/store_base.rb:31]: got some comments here

File: spec/win32/functional/win32/certstore_spec.rb [hunk 2]
```diff
@@ -30,44 +40,50 @@
...
       thumbprint = "b1bc968bd4f49d622aa89a81f2150152a41d829c"
-      expect(@store).to receive(:cert_get).with(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: "My").and_return(cert_pem)
-      @store.get(thumbprint, store_location: CERT_SYSTEM_STORE_CURRENT_USER, store_name: "My")
+      expect(@store).to receive(:cert_get).with(thumbprint).and_return(cert_pem)
+      @store.get(thumbprint)
     end
 
     # passing invalid thumbprint
-    it "returns nil if certificate not found" do
+    it "returns raises an Arugment error" do
```
Comment (lamont-granquist) [inline spec/win32/functional/win32/certstore_spec.rb:54]: is this a potentially breaking change to the API?

File: spec/win32/unit/certstore_spec.rb [hunk 3]
```diff
@@ -525,313 +479,202 @@
...
     end
 
-    context "When passing valid certificate's thumbprint" do
+    context "When passing a valid thumbprint with spaces or colons to the LocalMachine store" do
       let(:store_name) { "root" }
-      let(:store_location) { CERT_SYSTEM_STORE_CURRENT_USER }
-      let(:thumbprint) { "b1bc968bd4f49d622aa89a81f2150152a41d829909c" }
-      let(:cert_pem) { File.read('.\spec\win32\assets\GlobalSignRootCA.pem') }
+      let(:thumbprint) { "b1 bc 96 8b d4 f4 9d 62 2a a8 9a 81 f2 15 01 52 a4 1d 82 9c" }
+      let(:thumbprint2) { "b1:bc:96:8b:d4:f4:9d:62:2a:a8:9a:81:f2:15:01:52:a4:1d:82:9c" }
```
Comment (lamont-granquist) [inline spec/win32/unit/certstore_spec.rb:650]: are the changes to the thumbprint format here also a breaking change to the API?

**Comment by lamont-granquist** [comment]
It may be safer to bump the major version of this gem before shipping it.

File: lib/win32/certstore/store_base.rb (hunk 1 above)
Comment (johnmccrae) [reply lib/win32/certstore/store_base.rb:31]: Solved

File: spec/win32/functional/win32/certstore_spec.rb (hunk 2 above)
Comment (johnmccrae) [reply spec/win32/functional/win32/certstore_spec.rb:54]: I undid those to be more permissive about thumbprint format.

File: spec/win32/unit/certstore_spec.rb (hunk 3 above)
Comment (johnmccrae) [reply spec/win32/unit/certstore_spec.rb:646]: I undid those to be more permissive about thumbprint format.

File: Gemfile [hunk 4]
```diff
@@ -4,6 +4,7 @@ source "https://rubygems.org"
 gemspec
 
 gem "mixlib-shellout", "< 3.2.3"
+gem "chef-powershell", ">= 1.0.4"
```
Comment (tas50) [inline Gemfile:7]: if this is in the gemspec it doesn't need to be here

File: Gemfile [hunk 5]
```diff
@@ -4,6 +4,7 @@ source "https://rubygems.org"
...
 
 if Gem.ruby_version.to_s.start_with?("2.5")
```
Comment (tas50) [inline Gemfile:9]: There's ruby 2.5 logic here, but you removed the ruby 2.5 testing. If we support it we should test it. If not remove it all and update the dep in the gemspec

File: Gemfile (hunk 4 above)
Comment (johnmccrae) [reply Gemfile:7]: it's in the GEMFILE

File: Gemfile (hunk 5 above)
Comment (johnmccrae) [reply Gemfile:9]: The gemspec had a requirement in there already. Removed this.

File: Gemfile [hunk 6]
```diff
@@ -4,7 +4,6 @@ source "https://rubygems.org"
 gemspec
 
 gem "mixlib-shellout", "< 3.2.3"
```
Comment (tas50) [inline Gemfile]: Is there a reason we're pinning this to an older release?

Comment (johnmccrae) [reply Gemfile]: No?

Comment (johnmccrae) [reply Gemfile]: Killing that in the GEMFILE and the gemspec file.

File: Gemfile [hunk 7]
```diff
@@ -3,7 +3,7 @@ source "https://rubygems.org"
 # Specify your gem's dependencies in win32-certstore.gemspec
 gemspec
 
-gem "mixlib-shellout", "< 3.2.3"
+# gem "mixlib-shellout", "< 3.2.3"
```
Comment (lamont-granquist) [inline Gemfile:6]: this line should get removed if its no longer necessary

File: win32-certstore.gemspec [hunk 8]
```diff
@@ -19,7 +19,8 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency "bundler"
   spec.add_development_dependency "rspec", "~> 3.0"
 
-  spec.add_dependency "mixlib-shellout"
+  # spec.add_dependency "mixlib-shellout"
```
Comment (lamont-granquist) [inline win32-certstore.gemspec:22]: this should also get removed.

**Review by lamont-granquist — APPROVED** [review]

**Pull request merged**